data_structures/src/mainnet_validations.rs

```diff
@@ -100,6 +100,11 @@ impl TapiEngine {
 
         (min_epoch, old_wips)
     }
+
+    pub fn in_voting_range(&self, _epoch: Epoch, _wip: &str) -> bool {
+        // TODO: check if WIP accepts votes on this epoch
+        true
```

Comment:
for i in 0..tapi.len(){
    if let Some(bit_info) = tapi.get_mut(i, epoch) {
        if bit_info.wip == wip {
              return true;
        }
   }
}

false